samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -62,15 +65,19 @@
   private JobModel jobModel;
 
   public ZkJobCoordinator(int processorId, Config config, ScheduleAfterDebounceTime debounceTimer, ZkUtils zkUtils,
-      SamzaContainerController containerController, CoordinationUtils coordinationUtils) {
+      SamzaContainerController containerController) {
     this.zkUtils = zkUtils;
     this.keyBuilder = zkUtils.getKeyBuilder();
     this.debounceTimer = debounceTimer;
     this.processorId = processorId;
     this.containerController = containerController;
     this.zkController = new ZkControllerImpl(String.valueOf(processorId), zkUtils, debounceTimer, this);
     this.config = config;
-    this.coordinationUtils = coordinationUtils;
+    this.coordinationUtils = Util.
+        <CoordinationServiceFactory>getObj(
+            new JobCoordinatorConfig(config)
+                .getJobCoordinationServiceFactoryClassName())
+        .getCoordinationService("groupId", String.valueOf(processorId), config);
```

Comment:
Since this "groupId" will be coming from the creator of  SamzaProcessor, it needs to be passed into the ZkCoordinator.